src/rule_definition_tools.jl

```diff
@@ -198,6 +198,9 @@ function scalar_rrule_expr(f, call, setup_stmts, inputs, partials)
     end
 end
 
+"Declares properly hygenic inputs for propagation expressions"
+_propagator_inputs(n) = [esc(gensym(Symbol(:Δ, i))) for i in 1:n]
```

Comment:
Why do we need the `gensym`?

Reply:
`gensym` gives it a name that is certain not to clash with *anything*.
it is a name like `##Δ2#991`

So just incase the person had named a variable `Δ2` and written something like 
```
@scalar_rule foobar(x) (Δ2=10.0; Δ2*2),  x
```
using `gensym` makes sure we don't clash into that variable.

Reply:
ah yes `esc` doesn't help us with that